File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    
    mbti = {
        "RT": 0,
        "CF": 0,
        "JM": 0,
        "AN": 0,
    }
    
    for i in range(len(survey)) :
        if survey[i] in mbti:
            mbti[survey[i]] += choices[i] - 4
        else :
            survey[i] = survey[i][::-1]
            mbti[survey[i]] -= choices[i] - 4
        
    print(mbti)
    for e in mbti:
        list(e)
        if mbti[e] > 0 :
            answer += e[1]
        else :
            answer += e[0]
        
    return answer

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    let answer = "";
    let mbti = {
        "RT": 0,
        "CF": 0,
        "JM": 0,
        "AN": 0,
    }
    
    for(let i = 0; i < survey.length; i++){
        if(survey[i] in mbti){
            mbti[survey[i]] += choices[i] - 4;
        }else {
            survey[i] = survey[i].split("").reverse().join("");
            mbti[survey[i]] += -(choices[i] - 4);
        }
        console.log(mbti);
    }
    
    
    for(e in mbti){
        e.split("");
        console.log(e, mbti[e]);
        if (mbti[e] > 0) answer += e[1];
        else answer += e[0]
    }
    
    return answer;
}